### PR TITLE
Locking improvements earlier in chapter 12

### PR DESCRIPTION
**Pull request by chrishtr**
This PR:

* Removes some redundant code left over from previous PRs
* Adds a named setTimeout section
* Moves locking into the setTimeout section, where it is a natural addition and simplifies future discussions
* Gets rid of set_timeout as it was probably not adding any value, and may have been actively confusing as it has the same name as setTimeout
* Fixes various spelling errors

### CODE REVIEW

File: book/scheduling-and-threading.md [hunk 1]
```diff
@@ -278,11 +277,48 @@ class JSContext:
...
+the first thread and the timer thread run concurrently, and there is therefore
+no guarantee that one callback that writes to `schedule_task` will not be 
+interleaved with code on the other thread trying to read the task queue,
+leading to a [race condition](https://en.wikipedia.org/wiki/Race_condition)
+bug and nondeterministic results.
+
+This bug is easily fixed by use of a `threading.Lock` object. Before reading
+or writing to a data structure shared across threads, acquire the lock; after
+you're done, release it.^[The `blocking` parameter to `acquire` indicates
+whether the thread should block on acquiring the lock or not; in this chapter
```
Comment (pavpanchekha) [inline book/scheduling-and-threading.md:295]: I think we should define "block" here. As in,

> The `blocking` parameter to `acquire` indicates whether the thread should wait for the lock to be available; ...

File: book/scheduling-and-threading.md [hunk 2]
```diff
@@ -278,11 +277,48 @@ class JSContext:
...
+you'll always set it to true.] The code changes in `TaskRunner` are pretty
+easy---just be careful to not forget to release the lock, and hold it for the
+minimum time possible, so as to maximize thread parallelism.^[That's why the
+code releases the lock before calling `task`: after the task has been removed
+from the queue, it can't be accessed by another thread.]
```
Comment (pavpanchekha) [inline book/scheduling-and-threading.md:300]: I think the point about releasing before calling `task` is good and shouldn't be confined to a footnote.

**Review by pavpanchekha — APPROVED** [review]
You're right, this is a natural and effective place to talk about locking.

File: book/scheduling-and-threading.md (hunk 2 above)
Comment (chrishtr) [reply book/scheduling-and-threading.md]: Done.

**Pull request merged**